test/clojure/nrepl/transport_test.clj

```diff
@@ -9,3 +9,15 @@
       (is (thrown? IllegalArgumentException
                    (#'sut/safe-write-bencode out {"obj" (Object.)})))
       (is (empty? (.toByteArray out))))))
+
+(deftest tty-read-conditional-test
+  (testing "tty-read-msg is configured to read conditionals"
+    (let [in (-> "(try nil (catch #?(:clj Throwable :cljr Exception) e nil))"
+                 (java.io.StringReader.)
+                 (java.io.PushbackReader.))
+          out (ByteArrayOutputStream.)]
+      (is (= ['(try nil (catch Throwable e nil))]
+             (let [FnTransport (sut/tty in out nil)]
+               (.recv FnTransport)     ;; :op "clone"
```

Comment:
@bbatsov this wasn't as bad as I originally thought.
I saw eastwood fails on `.recv` saying "reference to field recv can't be resolved".  Any tip on how to resolve this?

Reply:
Probably you need to add a type signature there. Btw, I think the variable should be named `fn-transport`, not `FnTransport` as this looks like some type name. //cc @vemv

Reply:
There's a type `FnTransport` and also a defn `fn-transport` in nrepl.transport.  There must be a better name but I just named it `fn-transport` as suggested.